File: app/src/test/java/ru/ztrap/iconics/TestStringGenerator.java
package ru.ztrap.iconics;

import com.mikepenz.google_material_typeface_library.GoogleMaterial;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import javax.xml.parsers.ParserConfigurationException;
import javax.xml.transform.TransformerException;

/**
 * @author pa.gulko zTrap (29.03.2018)
 */
@RunWith(JUnit4.class)
public class TestStringGenerator extends IconicsStringGenerator {

    @Test
    public void generateGoogleMaterial() throws TransformerException, ParserConfigurationException {
        generateIconsFrom(new GoogleMaterial());
    }

    @Override
    protected FileCreationStrategy fileCreationStrategy() {
        return FileCreationStrategy.SAVE_ONLY_CURRENT;
    }
}

File: app/src/test/kotlin/ru/ztrap/iconics/TestStringGenerator.kt
package ru.ztrap.iconics

import com.mikepenz.iconics.typeface.library.googlematerial.GoogleMaterial
import org.junit.Ignore

import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import ru.ztrap.iconics.IconicsStringGenerator.FileCreationStrategy.SAVE_OLD

/**
 * @author pa.gulko zTrap (29.03.2018)
 */
@RunWith(JUnit4::class)
class TestStringGenerator : IconicsStringGenerator() {

    @Test fun generateGoogleMaterial() {
        GoogleMaterial.generateIcons()
    }

    override val fileCreationStrategy: FileCreationStrategy = SAVE_OLD
}
